app/assets/javascripts/miq_tree.js

```diff
@@ -503,11 +503,13 @@ window.miqInitTree = function(options, tree) {
     // Process nodes bottom-to-top
     while (parents.length > 0) {
       var node = parents.pop();
-      node.state.checked = node.nodes.map(function(node) {
-        return node.state ? node.state.checked : false;
-      }).reduce(function(acc, curr) {
-        return (acc === curr) ? acc : 'undefined';
-      });
+      if (node.nodes.length > 0) {
+        node.state.checked = node.nodes.map(function(node) {
+          return node.state ? node.state.checked : false;
+        }).reduce(function(acc, curr) {
+          return (acc === curr) ? acc : 'undefined';
+        });
+      }
```

Comment:
should we add an `else` case to set `node.state.checked` when there are no child nodes?

Reply:
I have no problem adding it. Although we don't render the checkbox if the node doesn't have any children, right?

Reply:
We don't alter the checkbox if there are no children.